driverprocessresults/main.py

```diff
@@ -0,0 +1,208 @@
+import os
+import shutil
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+from utils import *
+from driverprocessresults.scaling_plotter import *
+
+from pathlib import Path
+from statistics import median
+from configparser import ConfigParser
+
+from smartsim.log import get_logger, log_to_file
+logger = get_logger("Scaling Tests")
+
+
+class ProcessResults:
+    def process_scaling_results(self, 
+                                scaling_results_dir="aggregation-standard-scaling-py-fs", 
+                                overwrite=True):
+            """Create a results directory with performance data and plots
+            With the overwrite flag turned off, this function can be used
+            to build up a single csv with the results of runs over a long
+            period of time.
+            :param scaling_results_dir: directory to create results from
+            :type scaling_results_dir: str, optional
+            :param overwrite: overwrite any existing results
+            :type overwrite: bool, optional
+            """
+            dataframes = []
+            final_stat_dir = "results" / Path(scaling_results_dir) / "stats"
+            run_folders = os.listdir("results" / Path(scaling_results_dir))
+            session_folders = []
+            run_list = []
+            for run_folder in run_folders:
+                
+                if 'run' in run_folder:
+                    session_folders += ["results/" + scaling_results_dir + "/" + run_folder + "/" + d for d in os.listdir("results/" + scaling_results_dir + "/" + run_folder) if "sess" in d]
+                    run_list += [Path("results/" + scaling_results_dir + "/" + run_folder)]
+            try:
+                # write csv each so this function is idempotent
+                # csv's will not be written if they are already created
+                #tqdm creates a smart progress bar for the loops
+                for session_folder in tqdm(session_folders, desc="Processing scaling results...", ncols=80): #QUESTION: ncols: hardcoded?
```

Comment:
terminals tend to be 80 characters wide. Probably shouldn't have this comment in production code